airflow/provider.yaml.schema.json

```diff
@@ -195,11 +195,33 @@
     },
     "hook-class-names": {
       "type": "array",
-      "description": "Hook class names that provide connection types to core",
+      "description": "Hook class names that provide connection types to core (deprecated by connection-types)",
       "items": {
-        "type": "string"
+          "type": "string"
+      },
+      "deprecated": {
+        "description": "The hook-class-names property has been deprecated in favour of connection-types which is more performant version allowing to only import individual Hooks rather than all hooks at once",
+        "deprecatedVersion": "2.2"
       }
     },
+    "connection-types": {
+      "type": "array",
+      "description": "Map of connection types mapped to hook class names",
```

Comment:
```suggestion
      "description": "List of connection types mapped to hook class names",
```

or

```suggestion
      "description": "Array of connection types mapped to hook class names",
```

Reply:
@potiuk Was there a reason you have this as a list of dicts rather than just a map as originally documented?

i.e. instead of this currently

```yaml
connection-types:
  - hook-class-name: airflow.providers.airbyte.hooks.airbyte.AirbyteHook
    connection-type: airbyte
```

why didn't we have

```
connection-types:
  airbyte: hook-class-name: airflow.providers.airbyte.hooks.airbyte.AirbyteHook
```

I'm only now looking at this PR (sorry!) now to take over Daniel's `@task.docker` decorator work. and I'd like to follow the similar same pattern here.

I guess this is farily set in stone anyway as the providers with this pattern have been released haven't they?

Reply:
The Map was really a copy&paste victim not intention.

It could also work with Map - because we have no possibility to duplicate connection_type. But map does not give us much here,  the uniqueness has to be checked anyway for every connection type, because it is cross-provider and we just drop any connection_type that we already have registered. 

But one reason why it is "better" is how jsonschema works nicer with hints and validation when you edit it.

Compare:

```
    "connection-types": {
      "type": "array",
      "description": "Array of connection types mapped to hook class names",
      "items": {
          "type": "object",
          "properties": {
              "connection-type": {
                  "description": "Type of connection defined by the provider",
                  "type": "string"
              },
              "hook-class-name": {
                  "description": "Hook class name that implements the connection type",
                  "type": "string"
              }
          }
      },
```

with (that's how it would look like if we use map:


```
    "additional-extras": {
      "type": "object",
      "description": "Additional extras that the provider should have"
    },
```

I have my provider.yaml mapped to the schema in IntelliJ and it does not only display the hints but also it validates if the fields are of the "string" type. It's a  bit trivial, but It makes me want to be a bit more verbose to give a better hint/validation to users. 

![image](https://user-images.githubusercontent.com/595491/133659323-e6276c42-15de-4638-bfbf-bac47a93d00c.png)


![image](https://user-images.githubusercontent.com/595491/133659483-7a5a9500-a938-4ebd-a37a-854886626278.png)


Also I think it reads a bit better when you have array of (effectitvely) named tuples. It's more read-optimised than  write-optimised.

Reply:
And I think the "additional extras" was mistake BTW. It would be much better if I added it similar way as connection types.